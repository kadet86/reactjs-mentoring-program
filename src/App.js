import React, {Component, PureComponent} from 'react';

const noJsxElement = React.createElement('h3', {style: {color: "red"}}, 'Hello from no jsx element');

class HelloPure extends PureComponent {
    render() {
        return <h3 {...this.props}>Hello from pure component</h3>;
    }
}

function HelloFunc(props) {
    return <h3 {...props}>Hello from functional component</h3>;
}

export default class App extends Component {
    render() {
        return (
            <section>
                <h1>App Component header</h1>
                {noJsxElement}
                <HelloPure style={{color: "orange"}} />
                <HelloFunc style={{color: "green"}} />
            </section>
        );
    }
}
