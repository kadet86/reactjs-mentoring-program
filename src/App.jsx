import React from 'react';
import ErrorBoundary from './ErrorBoundary';
import FilteredMovieList from './FilteredMovieList';
import MoviePage from './MoviePage';

import { BrowserRouter as Router, Route, Redirect, Switch } from 'react-router-dom';

import './main.css';
import 'primereact/resources/themes/nova-light/theme.css';
import { NotFound } from './NotFound';

const App = () => {
    return (
        <ErrorBoundary>
            <main className="app">
                <Router>
                    <Switch>
                        <Route path="/" exact render={() => <Redirect to="/search" />} />
                        <Route path="/search/:query?/:searchBy?/:sortBy?" component={FilteredMovieList} />
                        <Route path="/film/:id" component={MoviePage} />
                        <Route component={NotFound} />
                    </Switch>
                </Router>
            </main>
        </ErrorBoundary>
    );
}

export default App;