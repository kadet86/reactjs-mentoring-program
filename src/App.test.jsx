import React from 'react';
import App from './App';
import {shallow} from 'enzyme';

describe('App component', () => {
    it('has persistent default snapshot', () => {
        const component = shallow(
            <App />,
        );
    
        expect(component).toMatchSnapshot();
    });

    it('has proper default state', () => {
        const component = shallow(
            <App />,
        );
    
        expect(component.state()).toEqual({movie: null});
    });

    it('has persistent snapshot after .navigateMovie() call', () => {
        const component = shallow(
            <App />,
        );

        component.instance().navigateToMovie({});
    
        expect(component).toMatchSnapshot();
    });

    it('has persistent snapshot after .navigateToSearch() call', () => {
        const component = shallow(
            <App />,
        );

        component.instance().navigateToSearch();
    
        expect(component).toMatchSnapshot();
    });
});

