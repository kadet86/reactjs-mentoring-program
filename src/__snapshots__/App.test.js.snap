// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot is persistent 1`] = `
<section>
  <h1>
    App Component header
  </h1>
  <h3
    style={
      Object {
        "color": "red",
      }
    }
  >
    Hello from no jsx element
  </h3>
  <h3
    style={
      Object {
        "color": "orange",
      }
    }
  >
    Hello from pure component
  </h3>
  <h3
    style={
      Object {
        "color": "green",
      }
    }
  >
    Hello from functional component
  </h3>
</section>
`;
